src/auth/sagas.test.js

```diff
@@ -1,95 +1,114 @@
-import { call, takeEvery, put } from 'redux-saga/effects'
-import authSaga, { loginSaga, logOutSaga, registerSaga } from './sagas'
-import { logIn as APILogIn, logOut as APILogOut, register as APIRegister, subscribeToAuthStateChanges } from './api'
-import { AuthActionType, logOutSuccess, loginSuccess, logIn, logOut, register } from './redux'
-import { toFakeUser } from './doubles'
+import { call, takeEvery, put } from 'redux-saga/effects';
+import authSaga, { loginSaga, logOutSaga, registerSaga } from './sagas';
+import {
+  logIn as APILogIn,
+  logOut as APILogOut,
+  register as APIRegister,
+  subscribeToAuthStateChanges,
+} from './api';
+import {
+  AuthActionType,
+  logOutSuccess,
+  loginSuccess,
+  logIn,
+  logOut,
+  register,
+} from './redux';
+import { toFakeUser } from './doubles';
 
 describe('authSaga', () => {
-
-  let gen
+  let gen;
 
   beforeEach(() => {
-    gen = authSaga()
-  })
+    gen = authSaga();
+  });
 
   it('should subscribe to changes to authed user in firebase', () => {
-    expect(gen.next().value).toEqual(call(subscribeToAuthStateChanges, loginSuccess, logOutSuccess))
-  })
+    expect(gen.next().value).toEqual(
+      call(subscribeToAuthStateChanges, loginSuccess, logOutSuccess),
+    );
+  });
 
   it('should take every logIn action and trigger loginSaga', () => {
-    gen.next()
-    expect(gen.next().value).toEqual(takeEvery(AuthActionType.LoginRequest, loginSaga))
-  })
+    gen.next();
+    expect(gen.next().value).toEqual(
+      takeEvery(AuthActionType.LoginRequest, loginSaga),
+    );
+  });
 
   it('should take every logOut action and trigger logOutSaga', () => {
-    gen.next()
-    gen.next()
-    expect(gen.next().value).toEqual(takeEvery(AuthActionType.LogOutRequest, logOutSaga))
-  })
+    gen.next();
```

Comment:
it's advisable not to skip over yields -- it's fine for now, but I think you should have a `describe` for each path flow and an `it` for each generator yield.

Reply:
Generators are definitely the weak point of my Javascript knowledge. So any help in that realm is much appreciated!